apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Snippets/SpannerConnectionSnippets.cs

```diff
@@ -36,5 +58,251 @@ public void CreateConnection()
+                            // If a transient fault occurs, this entire method is re-run.
+                            for (var i = 0; i < 5; i++)
+                            {
+                                cmd.Parameters["Key"].Value = Guid.NewGuid().ToString("N");
+                                cmd.Parameters["StringValue"].Value = $"StringValue{i}";
+                                cmd.Parameters["Int64Value"].Value = i;
+                                await cmd.ExecuteNonQueryAsync();
+                            }
+
+                            await transaction.CommitAsync();
+                        }
+                    }
+                });
+            // End sample
+        }
+
+#if !NETCOREAPP1_0
+        [Fact]
+        public async Task TransactionScopeAsync()
+        {
+            await _fixture.EnsureTestDatabaseAsync().ConfigureAwait(false);
+
+            // Sample: TransactionScopeAsync
+            // Additional: CreateInsertCommand
+            var retryPolicy = new RetryPolicy<SpannerFaultDetectionStrategy>(RetryStrategy.DefaultExponential);
+
+            await retryPolicy.ExecuteAsync(
+                async () =>
+                {
+                    using (var scope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
+                    {
+                        using (var connection =
+                            new SpannerConnection(
+                                $"Data Source=projects/{_projectId}/instances/{_instanceName}/databases/{_databaseName}"))
+                        {
+                            await connection.OpenAsync();
+
+                            var cmd = connection.CreateInsertCommand(
+                                "TestTable", new SpannerParameterCollection
+                                {
+                                    {"Key", SpannerDbType.String},
+                                    {"StringValue", SpannerDbType.String},
+                                    {"Int64Value", SpannerDbType.Int64}
+                                });
+
+                            // This executes a single transactions with alls row written at once during scope.Complete().
+                            // If a transient fault occurs, this entire method is re-run.
+                            for (var i = 0; i < 5; i++)
+                            {
+                                cmd.Parameters["Key"].Value = Guid.NewGuid().ToString("N");
+                                cmd.Parameters["StringValue"].Value = $"StringValue{i}";
+                                cmd.Parameters["Int64Value"].Value = i;
+                                await cmd.ExecuteNonQueryAsync();
+                            }
+
+                            scope.Complete();
+                            await Task.Yield();
+                        }
+                    }
+                });
+            // End sample
+            //TODO(benwu): Remove Task.Yield once grpc# fixes threadpool issue.
```

Comment:
We're using Grpc.Core 1.6.1 now, so if we're thinking about the same thread pool issue, you may be able to remove it now.